vllm/worker/model_runner.py

```diff
@@ -49,7 +50,10 @@ def __init__(
         # FIXME(woosuk): This is a hack to make the tests work. Refactor this.
         self.sliding_window = (model_config.get_sliding_window()
                                if model_config is not None else None)
-        self.device = torch.device(torch.cuda.current_device())
+        self.device_config = (device_config
+                              if device_config is not None else DeviceConfig())
+        self.device = self.device_config.device
```

Comment:
BTW, I intentionally avoided using `torch.set_current_device`, since this can affect the user code when using `LLM` class.